tests/debugger/test_union.cpp

```diff
@@ -0,0 +1,40 @@
+#include <vector>
+#include <set>
+#include <map>
+#include <assert.h>
+#include <math.h>
+#include <string.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+#include <emscripten.h>
+
+union TestBase {
+    char c;
+    int i;
+    float f;
+};
+
+void test_1() {
+  TestBase tb;
+  tb.f = 13.37;
+
+  EM_ASM_INT({
+    var decoded = Module['emdebugger_heap_printer'].decode_from_stack($0, "tb", 100)["union TestBase"];
```

Comment:
No asserts, this test is still a WIP, though running it directly from a JS_ENGINE will give an idea of what the output looks like